src/main/java/net/neoforged/neoforge/network/registration/NetworkRegistry.java

```diff
@@ -0,0 +1,959 @@
+                return null;
+            }
+
+            final PlayRegistration<?> registration = knownPlayRegistrations.get(channel.id());
+            if (registration == null) {
+                LOGGER.error("Received a modded custom payload packet {} with an unknown or not accepted channel. Not parsing packet.", channel.id());
+                throw new IllegalStateException("A client send a packet with an unknown or not accepted channel, while negotiation succeeded. Somebody changed the channels known to NeoForge!");
+            }
+
+            if (registration.flow().isPresent()) {
+                if (registration.flow().get() != flow) {
+                    LOGGER.warn("Received a modded custom payload packet {} that is not supposed to be send to the server. Disconnecting client.", channel.id());
+                    final Connection connection = ConnectionUtils.getConnection(context);
+                    final PacketListener listener = connection.getPacketListener();
+                    if (listener instanceof ServerGamePacketListener serverListener) {
+                        serverListener.disconnect(Component.translatable("neoforge.network.invalid_flow", flow));
+                    } else if (listener instanceof ClientGamePacketListener clientListener) {
+                        clientListener.getConnection().disconnect(Component.translatable("neoforge.network.invalid_flow", flow));
+                    } else {
+                        LOGGER.error("Received a modded custom payload packet {} that is not supposed to be send to the server. Disconnecting client, but the listener is not a game listener. This should not happen.", channel.id());
+                        throw new IllegalStateException("A client send a packet with an unknown or not accepted channel, while negotiation succeeded. Somebody changed the channels known to NeoForge!");
+                    }
+                }
+            }
+
+            return registration;
+        } else if (protocol.isConfiguration()) {
+            final NetworkChannel channel = payloadSetup.configuration().get(id);
+
+            //Also validate that everything is key and then return a reader.
+            if (channel == null) {
+                LOGGER.warn("Received a modded custom payload packet {} with an unknown or not accepted channel. Not parsing packet.", id);
+                return null;
+            }
+
+            final ConfigurationRegistration<?> registration = knownConfigurationRegistrations.get(channel.id());
+            if (registration == null) {
+                LOGGER.error("Received a modded custom payload packet {} with an unknown or not accepted channel. Not parsing packet.", channel.id());
+                throw new IllegalStateException("A client send a packet with an unknown or not accepted channel, while negotiation succeeded. Somebody changed the channels known to NeoForge!");
```

Comment:
`sent` not `send`. Also, that assumption in the exception sounds.. wrong and should be removed as it jumps to conclusions.

Reply:
The assumption is correct. There is no other systems that makes these registration objects, and as such an external influence modified them.